lib/features/context-pad/ContextPad.js

```diff
@@ -507,12 +510,20 @@ ContextPad.prototype.isShown = function() {
 /**
  * Get contex pad position.
  *
+ * If target is a connection, the context pad will be placed according to the
+ * cnnection's last waypoint.
+ *
+ * If multiple targets, the context pad will be placed according to the bounding
+ * box containing all targets.
+ *
  * @param {ContextPadTarget} target
  *
  * @return {Rect}
  */
 ContextPad.prototype._getPosition = function(target) {
 
+  target = isConnection(target) ? getLastWaypoint(target) : target;
```

Comment:
What if I pass a single-item array with connection?

Reply:
So `getPad([ connection ]);`

Reply:
We check for that here: https://github.com/bpmn-io/diagram-js/blob/0cfd9c77ca823f6eac4d549b95e1902817536893/lib/features/context-pad/ContextPad.js#L111

Reply:
OK but that code is not run when I open context pad programmatically. I know that we are facing legacy logic here. Notice how we unwrap the target from array to wrap it again in `_getPosition`.

Reply:
It looks like we don't handle programmatic opening of a single item array anyway, so perhaps we could refactor this context pad in a separate PR (some day maybe).

Reply:
True, programatically it doesn't really make sense